src/inpod/workloadmanager.rs

```diff
@@ -184,6 +192,9 @@ impl WorkloadProxyManager {
             };
             debug!("workload proxy manager is NOT ready");
             self.readiness.not_ready();
+            backoff = std::cmp::min(MAX_BACKOFF, backoff * 2);
```

Comment:
We may need some way to reset the backoff. Otherwise, eventually over a ztunnel running for a long time it will always hit the 15s backoff (?)

Reply:
If there are sustained protocol errors we'll never get marked as ready/will stay unready until k8s kills us. Which is about all we can do.

If there are intermittent/transient protocol errors that then somehow resolve themselves(?) yep, though that's not likely to manifest as anything terribly severe or happen.

I can tie the backoff reset to the readiness toggle, so it resets when we go ready.

Reply:
I was thinking like ztunnel runs for a month and cni restarts daily or something. Not sure how realistic that is though

Reply:
> I was thinking like ztunnel runs for a month and cni restarts daily or something. Not sure how realistic that is though

That would be handled by the inner socket backoff, and we wouldn't get to this point.

This backoff is just for protocol errors, not transport errors.

Reply:
Actually, maybe we should just die on protocol errors. Probably over-complicating this.